ib-cli/installationiso/installationiso.go

```diff
@@ -0,0 +1,273 @@
+	r.log.Info("Creating IBI installation ISO")
+	err := r.validate()
+	if err != nil {
+		return err
+	}
+	err = r.createIgnitionFile(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk)
+	if err != nil {
+		return err
+	}
+	if err := r.downloadLiveIso(rhcosLiveIsoUrl); err != nil {
+		return err
+	}
+	if err := r.embedIgnitionToIso(); err != nil {
+		return err
+	}
+	r.log.Infof("installation ISO created at: %s", path.Join(r.workDir, ibiIsoFileName))
+
+	return nil
+}
+
+func (r *InstallationIso) validate() error {
+	_, err := os.Stat(r.workDir)
+	if err != nil && os.IsNotExist(err) {
+		return fmt.Errorf("work dir doesn't exists %w", err)
+	}
+	return err
+}
+
+func (r *InstallationIso) createIgnitionFile(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk string) error {
+	r.log.Info("Generating Ignition Config")
+	err := r.renderButaneConfig(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk)
+	if err != nil {
+		return err
+	}
+	return r.renderIgnitionFile()
+}
+
+func (r *InstallationIso) renderIgnitionFile() error {
+	ibiIsoPath := path.Join(r.workDir, ibiIgnitionFileName)
+	if _, err := os.Stat(ibiIsoPath); err == nil {
+		r.log.Infof("ignition file exists (%s), deleting it", ibiIsoPath)
+		if err = os.Remove(ibiIsoPath); err != nil {
+			return fmt.Errorf("failed to delete existing ignition config: %w", err)
+		}
+	}
+
+	command := "podman"
+	args := []string{"run",
+		"-v", fmt.Sprintf("%s:/data:rw,Z", r.workDir),
+		"--rm",
+		"quay.io/coreos/butane:release",
+		"--pretty", "--strict",
+		"-d", "/data",
+		path.Join("/data", butaneConfigFile),
+	}
+	ignitionContent, err := r.ops.RunInHostNamespace(command, args...)
+	if err != nil {
+		return fmt.Errorf("failed to render ignition from config: %w", err)
+	}
+	return os.WriteFile(path.Join(r.workDir, ibiIgnitionFileName), []byte(ignitionContent), 0o644)
+}
+
+func (r *InstallationIso) embedIgnitionToIso() error {
+	ibiIsoPath := path.Join(r.workDir, ibiIsoFileName)
+	if _, err := os.Stat(ibiIsoPath); err == nil {
+		r.log.Infof("ibi ISO exists (%s), deleting it", ibiIsoPath)
+		if err = os.Remove(ibiIsoPath); err != nil {
+			return fmt.Errorf("failed to delete existing ibi ISO: %w", err)
+		}
+	}
+
+	command := "podman"
+	args := []string{"run",
+		"-v", fmt.Sprintf("%s:/data:rw,Z", r.workDir),
+		coreosInstallerImage,
+		"iso", "ignition", "embed",
+		"-i", path.Join("/data", ibiIgnitionFileName),
+		"-o", path.Join("/data", ibiIsoFileName),
+		path.Join("/data", rhcosIsoFileName),
+	}
+
+	if _, err := r.ops.RunInHostNamespace(command, args...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *InstallationIso) renderButaneConfig(seedImage, seedVersion, authFile, pullSecretFile, sshPublicKeyPath, lcaImage, installationDisk string) error {
+	r.log.Debug("Generating butane config")
+	ignitionTemplate, err := folder.ReadFile(ibiButaneTemplateFilePath)
```

Comment:
You can use RenderTemplate from utils.go

Reply:
Fixed